packages/web/src/pages/search-page-v2/SearchFilters.tsx

```diff
@@ -181,31 +194,119 @@ const KeyFilter = () => {
   )
 }
 
-// TODO: Need to debounce the on change for this bc it locks up the UI a bit. only like 100ms
+type BpmTargetType = 'exact' | 'range5' | 'range10'
+const targetOptions: { label: string; value: BpmTargetType }[] = [
+  {
+    label: 'Exact',
+    value: 'exact'
+  },
+  {
+    label: '± 5',
+    value: 'range5'
+  },
+  {
+    label: '± 10',
+    value: 'range10'
+  }
+]
+const bpmOptions = [
+  {
+    label: 'Very Slow',
+    value: '1-60',
+    helperText: '<60 BPM'
+  },
+  {
+    label: 'Slow',
+    value: '60-90',
+    helperText: '60-90 BPM'
+  },
+  {
+    label: 'Medium',
+    value: '90-110',
+    helperText: '90-110 BPM'
+  },
+  {
+    label: 'Upbeat',
+    value: '110-140',
+    helperText: '110-140 BPM'
+  },
+  {
+    label: 'Fast',
+    value: '140-160',
+    helperText: '140-160 BPM'
+  },
+  {
+    label: 'Very Fast',
+    value: '160-999',
+    helperText: '160+ BPM'
+  }
+]
+
+// TODO: Need to debounce the on change for this when user is typing. it locks up the UI a bit. only like 100ms
+// Jank:
+// - When bpm is set in target view and then an option is clicked in the range view, it clears on first click
+// -
 const BpmFilter = () => {
+  const { color } = useTheme()
   const [urlSearchParams] = useSearchParams()
   const bpm = urlSearchParams.get('bpm')
   const updateSearchParams = useUpdateSearchParams('bpm')
-
   const [bpmFilterType, setBpmFilterType] = useState<'range' | 'target'>(
     'range'
   )
+  const [bpmTargetType, setBpmTargetType] = useState<BpmTargetType>('exact')
 
-  const label = bpm ? `${bpm} ${messages.bpm}` : `${messages.bpm}`
+  const minRef = useRef('')
+  const maxRef = useRef('')
+  const targetBpmRef = useRef('')
+  const targetTypeRef = useRef<BpmTargetType>('exact')
 
-  const handleBpmInputChange = useCallback(
-    (handleChange: (value: string, label: string) => void) =>
-      (value: string) => {
-        handleChange(value, `${value} ${messages.bpm}`)
-      },
+  const handleRangeChange = useCallback(
+    (handleChange: (value: string, label: string) => void) => {
+      let value = ''
+
+      if (minRef.current || maxRef.current) {
+        value = `${minRef.current || '1'}-${maxRef.current || '999'}`
+      }
+
+      handleChange(value, `${value} ${messages.bpm}`)
+    },
+    []
+  )
+
+  const handleTargetChange = useCallback(
+    (handleChange: (value: string, label: string) => void) => {
```

Comment:
ideally this should be the "ChangeEvent" type so it actually acts like an input field, prob worth a discussion with the original implementor of filterbutton